src/AbstractContainer.php

```diff
@@ -192,7 +192,7 @@ protected function verifyNamespace($createContainer = true)
     {
         $storage = $this->getStorage();
         $name    = $this->getName();
-        if (!isset($storage[$name])) {
+        if (!is_array($storage[$name]) && !$storage[$name] instanceof Traversable) {
```

Comment:
The `isset()` is still necessary, and should be performed before the other checks.